### DISCUSSION TITLE
Color and sort by file status

### DISCUSSION THREAD

**Discussion by stanio** [jesseduffield/lazygit#3406] [Ideas]
As a long-time [Sourcetree](https://www.sourcetreeapp.com/) user (and a fugitive – it's just too slow) I'm used to its status coloring and sorting:

![Sourcetree Window – Pending files, sorted by file status](https://github.com/jesseduffield/lazygit/assets/1247730/5e0cdc74-52e2-40e9-9185-9917d6dd7b9d)

In LazyGit I'm using the flat file list view (<kbd>`</kbd> Toggle file tree view) and I would like the ability to:

-   Configure colors for the different file statuses
-   Sort the files like:

    -   Staged
        -   Modified
        -   Renamed
        -   Added
        -   Deleted
    -   Unstaged
        -   Modified
        -   Missing (Deleted)
        -   Untracked

I often have lots of unstaged and untracked changes, so having the modified files "pop up" at the top is pretty much crucial to me.  Then, having any/all staged changes "pop up" at the very top of the list is necessary for me to know what I'm about to commit.

Does anyone find such setup useful?

**Comment by yonilevy** [comment]
Yes!

The minimal version of this would be a new config option, say: `filesFlatSortOrder` with options `(byName, byStatus)`, where `byStatus` orders files by (status, name) with status order (staged, unstaged, untracked). This might only make sense with `showFileTree=false`, hence the *flat* sort order.

Actually, it might be better to consolidate both configs to a more general `filesDisplayMode` with options `(tree, flat, flatByStatus)`...

I believe this was also the initial ask in #988 but it ended up being implemented as a filter, which I find much less useful.
